src/login/LoginPage.jsx

```diff
@@ -1,395 +1,318 @@
-          <Form name="sign-in-form" id="sign-in-form">
-            <FormGroup
-              name="emailOrUsername"
-              value={this.state.emailOrUsername}
-              autoComplete="on"
-              handleChange={(e) => this.setState({ emailOrUsername: e.target.value, isSubmitted: false })}
-              handleFocus={this.handleOnFocus}
-              handleBlur={this.handleOnBlur}
-              errorMessage={this.state.errors.emailOrUsername}
-              floatingLabel={intl.formatMessage(messages['login.user.identity.label'])}
-            />
-            <PasswordField
-              name="password"
-              value={this.state.password}
-              autoComplete="off"
-              showRequirements={false}
-              handleChange={(e) => this.setState({ password: e.target.value, isSubmitted: false })}
-              handleFocus={this.handleOnFocus}
-              handleBlur={this.handleOnBlur}
-              errorMessage={this.state.errors.password}
-              floatingLabel={intl.formatMessage(messages['login.password.label'])}
-            />
-            <StatefulButton
-              name="sign-in"
-              id="sign-in"
-              type="submit"
-              variant="brand"
-              className="login-button-width"
-              state={submitState}
-              labels={{
-                default: intl.formatMessage(messages['sign.in.button']),
-                pending: '',
-              }}
-              onClick={this.handleSubmit}
-              onMouseDown={(e) => e.preventDefault()}
+          <PasswordField
+            name="password"
+            value={formFields.password}
+            autoComplete="off"
+            showScreenReaderText={false}
+            showRequirements={false}
+            handleChange={handleOnChange}
+            handleFocus={handleOnFocus}
+            errorMessage={errors.password}
+            floatingLabel={formatMessage(messages['login.password.label'])}
+          />
+          <StatefulButton
+            name="sign-in"
+            id="sign-in"
+            type="submit"
+            variant="brand"
+            className="login-button-width"
+            state={submitState}
+            labels={{
+              default: formatMessage(messages['sign.in.button']),
+              pending: '',
+            }}
+            onClick={handleSubmit}
+            onMouseDown={(event) => event.preventDefault()}
+          />
+          <Link
+            id="forgot-password"
+            name="forgot-password"
+            className="btn btn-link font-weight-500 text-body"
+            to={updatePathWithQueryParams(RESET_PAGE)}
+            onClick={trackForgotPasswordLinkClick}
+          >
+            {formatMessage(messages['forgot.password'])}
+          </Link>
+          {provider && (
+            <EnterpriseSSO provider={provider} />
+          )}
+          {!provider && (
```

Comment:
Why is `!provider` condition here? In prev implementation the whole form was rendered `renderForm` when there was no `provider` not only the third party auth.

I guess its also related to my comment

> in the prev implementation, the EnterpriseSSO component was only being rendered in the if (this.tpaHint) and then the provider conditional, this is not true in here. Its only under the provider conditional here.

i think either I have got confused or we may have misunderstood the conditionals here.

Reply:
Resolving it since its related to https://github.com/openedx/frontend-app-authn/pull/899#discussion_r1458554946